### PR TITLE
*: replace failure with anyhow and thiserror

### PR DESCRIPTION
**Pull request by koushiro**
### What problem does this PR solve?

Problem Summary: part of #8698

### What is changed and how it works?

What's Changed: replace `failure` with `anyhow` and `thiserror` in some crates

### Check List 

Tests 

- No code

Side effects

- No

### Release note

No

### CODE REVIEW

**Review by nrc — APPROVED** [review]
lgtm, thanks!

File: components/backup/src/lib.rs [hunk 1]
```diff
@@ -3,8 +3,6 @@
 #![recursion_limit = "200"]
```
Comment (BusyJay) [inline components/backup/src/lib.rs:3]: Is it still necessary?

File: components/tidb_query_common/Cargo.toml [hunk 2]
```diff
@@ -17,8 +17,8 @@ prost-codec = [
 ]
 
 [dependencies]
+anyhow = "1.0"
```
Comment (BusyJay) [inline components/tidb_query_common/Cargo.toml:20]: Is it possible to keep only depending on thiserror?

Comment (nrc) [reply components/tidb_query_common/Cargo.toml:20]: I think we need anyhow for dynamic errors. IMO, we shouldn't have any dynamic errors, but in the short term, I think it is easier to keep using them and depend on anyhow. Longer term I have some ideas to transition away.

File: components/backup/src/lib.rs (hunk 1 above)
Comment (koushiro) [reply components/backup/src/lib.rs]: no, I will remove it

**Review by BusyJay — APPROVED** [review]

**Pull request merged**